### PR TITLE
Remove global chart envs

### PR DESCRIPTION
**Pull request by tiina303**
## Description

Users can now add additional global envs into their values.yaml like this
```
env:
  - name: ADDITIONAL_ENV
    value: "false"
```
Instead of
```
env:
  - name: ASYNC_EVENT_PROPERTY_USAGE
    value: "true"
  - name: EVENT_PROPERTY_USAGE_INTERVAL_SECONDS
    value: "86400"
  - name: ADDITIONAL_ENV
    value: "false"
```

Depends on the PostHog version in use having this change https://github.com/PostHog/posthog/pull/7709
Fortunately that's the case for another PR already with a similar requirement: https://github.com/PostHog/charts-clickhouse/pull/179#issuecomment-1009872485

Opening as draft to make sure we don't land it too soon. Also made it a patch release with the assumption that we land it after 179 (which will be the major with PostHog version requirement).

## Type of change
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## How has this been tested?

Will run CI e2e tests before landing.

### CODE REVIEW

**Comment by guidoiaquinti** [comment]
I'm not sure I'm following the purpose of this PR. Is the plan to put all the defaults in `posthog/posthog` and remove the overrides here?

**Comment by tiina303** [comment]
> I'm not sure I'm following the purpose of this PR. Is the plan to put all the defaults in posthog/posthog and remove the overrides here?

yes, I did that in the PR linked. The purpose is (1) making it easier/clearer for users how they can add overrides to envs. Sure event without this PR you could do 
```
env:
  - name: ADDITIONAL_ENV
    value: "false"
```
As the defaults are right in posthog/posthog, but I wouldn't assume I could do that looking at the `values.yaml`. So reducing confusion. Additionally (2) if we ever want to change any defaults for PostHog that should be in the code there as we can't easily tie chart version to PostHog version (I mean there are no guarantees), so this removes a bad pattern.

File: charts/posthog/values.yaml [hunk 1]
```diff
@@ -31,11 +31,7 @@ clickhouseOperator:
   serviceType: NodePort
 
 # -- Env vars to throw into every deployment (web, worker, and plugin server)
-env:
-  - name: ASYNC_EVENT_PROPERTY_USAGE
-    value: "true"
-  - name: EVENT_PROPERTY_USAGE_INTERVAL_SECONDS
-    value: "86400"
+env: []
```
Comment (fuziontech) [inline charts/posthog/values.yaml:34]: I would add a comment here to show how to add env vars if they want to. The [] is a bit confusing.

Just an example

```yaml
# env:
#   - name: FOO
#     value: bar
```

**Review by fuziontech — APPROVED** [review]
:shipit: but just add the comment (mentioned in the comment)

I guess that's a meta comment or higher order comment? :P

File: charts/posthog/values.yaml (hunk 1 above)
Comment (tiina303) [inline charts/posthog/values.yaml:34]:
```suggestion
env: []
# env: 
#   - name: FOO 
#     value: bar
```

Comment (tiina303) [reply charts/posthog/values.yaml:34]: Note that we do have an example later in the file now, but added since I though about it earlier too

**Comment by tiina303** [comment]
Ran the e2e test just in case too [AWS](https://github.com/PostHog/charts-clickhouse/actions/runs/1922189625), [DO](https://github.com/PostHog/charts-clickhouse/actions/runs/1922192879), [GCP](https://github.com/PostHog/charts-clickhouse/actions/runs/1922191622)

**Pull request merged**